### PR TITLE
Feature/sort ssh known host file

### PR DESCRIPTION
**Pull request by shrikeh**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,9 +1,24 @@
 source 'https://rubygems.org'
 
+<<<<<<< HEAD
```
Comment (andytson-inviqa) [inline Gemfile]: git conflict partially committed here

File: Gemfile [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
+  gem 'berkshelf', '~> 3.3.0'
+  gem 'knife-solo'
+  gem 'knife-solo_data_bag'
+
+  # Lock nokogiri to the version shipped with chefdk,
+  # to avoid installing it again
+  gem 'nokogiri', '< 1.6.6.3'
+end
+
+source 'https://rubygems.org'
```
Comment (andytson-inviqa) [inline Gemfile]: duplicate source

File: libraries/ssh_known_hosts.rb [hunk 3]
```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env ruby
+
+module SshKnownHosts
```
Comment (andytson-inviqa) [inline libraries/ssh_known_hosts.rb]: should this be SshUserKnownHosts now?

File: test/fixtures/cookbooks/fake/metadata.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#/usr/bin/env ruby
+# /usr/bin/env ruby
```
Comment (andytson-inviqa) [inline test/fixtures/cookbooks/fake/metadata.rb]: intentional? most of my bin interpreters don't have a space here

File: providers/entries.rb [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  @current_resource.mode(@new_resource.mode)
+end
+
+# Helper module for Chef
+module CreateUserKnownHostEntries
+  include SshKnownHosts
+
+  module_function
+
+  @regex_entry = Regexp.new('([^\s]+) ([a-z0-9-]+) ([0-9A-Za-z/+]+[=]+)')
```
Comment (andytson-inviqa) [inline providers/entries.rb]: ([0-9A-Za-z/+]+[=]+) needs to be ([0-9A-Za-z/+]+[=]*)

A base64 encoded string uses = for padding when a the decoded string isn't a multiple of 4. If it's a multiple of 4 then there is no padding

Comment (andytson-inviqa) [inline providers/entries.rb]: sorry \* multiple of 3

Comment (andytson-inviqa) [inline providers/entries.rb]: e.g.
1 - MQ==
12 - MTI=
123 - MTIz

File: libraries/ssh_known_hosts.rb [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  # Describes a host (IP, domain) for a host entry
+  class Host
+    @regex_ip_address = Regexp.new('\d+\.\d+\.\d+\.\d+')
```
Comment (andytson-inviqa) [inline libraries/ssh_known_hosts.rb]: note ipv6 ips may be present

File: libraries/ssh_known_hosts.rb [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    def to_s
+      host = domains + ips
+      host.join(',')
+    end
+  end
+
+  # Host key class for a value object describing a known_host key (type key)
+  class HostKey
+    def self.from_parts(key, type)
+      type = "ssh-#{type}" if type == 'rsa' || type == 'dsa'
```
Comment (andytson-inviqa) [inline libraries/ssh_known_hosts.rb]: the following seem to be the map
dsa - ssh-dss
ecdsa - ecdsa-sha2-nistp256 (on ubuntu)
rsa - ssh-rsa
ed25519 - ssh-ed25519
rsa1 - unknown, it didn't want to generate one

File: Gemfile (hunk 1 above)
Comment (marcomc) [inline Gemfile]: fixed

File: Gemfile (hunk 2 above)
Comment (marcomc) [reply Gemfile]: fixed

File: Gemfile (hunk 1 above)
Comment (marcomc) [reply Gemfile]: fixed

File: libraries/ssh_known_hosts.rb (hunk 3 above)
Comment (marcomc) [reply libraries/ssh_known_hosts.rb]: file and module now renamed to `KnownHosts`, seems correct to me
or should be named after the new name of the repo ssh_known_hosts_ng (SshKnownHostsNG)?
@andytson-inviqa

File: test/fixtures/cookbooks/fake/metadata.rb (hunk 4 above)
Comment (marcomc) [reply test/fixtures/cookbooks/fake/metadata.rb]: fixed

File: libraries/ssh_known_hosts.rb (hunk 3 above)
Comment (andytson-inviqa) [reply libraries/ssh_known_hosts.rb]: @marcomc yep that'd make sense renaming to match the name of the cookbook

File: providers/entries.rb (hunk 5 above)
Comment (marcomc) [reply providers/entries.rb]: fixed and moved to `libraries/known_hosts_file`

File: libraries/ssh_known_hosts.rb (hunk 7 above)
Comment (marcomc) [reply libraries/ssh_known_hosts.rb]: @andytson-inviqa he seems to have reduced the options to only `rsa`
https://github.com/shrikeh/chef-ssh-user-known-hosts/pull/3/files#diff-2337bdead6fc65d0020bb3221b0be0bfR60should we implement (now) all of the other types or we could leave that for later?

File: libraries/ssh_known_hosts.rb (hunk 6 above)
Comment (marcomc) [reply libraries/ssh_known_hosts.rb]: it seems it's been remove

**Pull request merged**